sklearn/utils/_get_params_html_repr.py

```diff
@@ -0,0 +1,98 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+import html
+import pprint
+from collections import UserDict
+
+from sklearn._config import get_config
+
+
+class ParamsDict(UserDict):
+    """Dictionary-like class to visualize parameters in HTML.
+
+    It builds an HTML structure to be used with Jupyter notebooks or similar
+    environments. It allows storing metadata to track non-default parameters.
+    """
+
+    def __init__(self, params=None, non_default=tuple()):
+        super().__init__(params or {})
+        self.non_default = non_default
+
+    @property
+    def _repr_html_(self):
+        # Taken from sklearn.base.BaseEstimator
+        """HTML representation of estimator.
+
+        This is redundant with the logic of `_repr_mimebundle_`. The latter
+        should be favored in the long term, `_repr_html_` is only
+        implemented for consumers who do not interpret `_repr_mimbundle_`.
+        """
+        if get_config()["display"] != "diagram":
+            raise AttributeError(
+                "_repr_html_ is only defined when the "
+                "'display' configuration option is set to "
+                "'diagram'"
+            )
+        return self._repr_html_inner
+
+    def _repr_html_inner(self):
+        return _html_template(self)
+
+    def _repr_mimebundle_(self, **kwargs):
+        """Mime bundle used by jupyter kernels to display estimator"""
+        output = {"text/plain": repr(self)}
+        if get_config()["display"] == "diagram":
+            output["text/html"] = _html_template(self)
+
+
+def _read_params(param, value, non_default_params, row_template):
+
+    if value != "deprecated" and isinstance(value, str):
+        cleaned_value = f'"{value}"'
+    else:
+        cleaned_value = html.escape(str(value))
+    if len(cleaned_value) > 50:
+        if param == "param_distributions":
+            formatted_value = pprint.pformat(cleaned_value)
+            cleaned_value = f"<pre>{formatted_value}</pre>"
+        else:
+            cleaned_value = "(...)"
```

Comment:
I see that the standard library have a nice tools that will be better because it will keep the external list symbol. It is the `reprlib`:

```python
            import reprlib
            r = reprlib.Repr()
            r.maxlist = 2      # Show only first 2 items of lists
            r.maxtuple = 1     # Show only first item of tuples
            r.maxstring = 8    # Limit string length to 8 chars
            cleaned_value = r.repr(value)
```

Depending on the parameter, it will automatically rework the repr for us. So I find it quite nice. For instance, with the code above, I get the following:

![image](https://github.com/user-attachments/assets/69598548-ff51-490f-afc9-975399b69a6d)